message c2sSignup
{
	enum MsgID
	{
		id = 1;
	}
	required string account_name = 1;
	required string password = 2;
}

message s2cSignup
{
	enum MsgID
	{
		id = 2;
	}
	
	required bool result = 1;
}

message c2sLogin
{
	enum MsgID
	{
		id = 3;
	}

	required string account_name=1;
	required string password=2;
}

message s2cLogin
{
	enum MsgID
	{
		id = 4;
	}

	required bool result=1;
}

message c2sCreateRole
{
	enum MsgID
	{
		id = 5;
	}
	
	required string role_name = 1;
}

message RoleBase
{
	required int64 role_id = 1;
	required string role_name = 2;
	optional int32 title = 3;
}

message RoleInfo
{
	optional int32 hp = 1;
	optional int32 mp = 2;
	optional int32 damage = 3;
	optional int32 defend = 4;
}

message s2cCreateRole
{
	enum MsgID
	{
		id = 6;
	}
	
	required bool result = 1;
	optional RoleBase base = 2;
}

message c2sSelectRole
{
	enum MsgID
	{
		id = 7;
	}
	
	required int64 role_id = 1;
}

message s2cSelectRole
{
	enum MsgID
	{
		id = 8;
	}
	
	required bool result = 1;
	optional RoleInfo info = 2;
}